### PR TITLE
#131/Add a fullName field to the User

### LINKED ISSUE #131: Add a fullName field to the User
- [x] Edit the `Users` table
- [x] Edit the `User` and `UserEntity`
- [x] Edit the `UserDto` and `UpdateUserDto`

### PR DESCRIPTION
**Pull request by gbudau**
Because we don't have migration scripts yet, please run `docker compose down -v` to remove the database volume

close #131

### CODE REVIEW

**Comment by Damrod** [comment]
Please, add this new friend to the User page in webapp as well

File: db/setup/schema.sql [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
-  "email" VARCHAR(50) NOT NULL UNIQUE,
-  "avatarId" UUID REFERENCES LocalFile(id) UNIQUE,
-  "createdAt" TIMESTAMPTZ NOT NULL DEFAULT NOW()
-);
+CREATE TABLE
+  IF NOT EXISTS Users(
+    "id" UUID DEFAULT gen_random_uuid() PRIMARY KEY,
+    "username" VARCHAR(20) NOT NULL UNIQUE,
+    "email" VARCHAR(50) NOT NULL UNIQUE,
+    "fullName" VARCHAR(255) NOT NULL,
```
Comment (gbudau) [inline db/setup/schema.sql:16]: I formatted this file with [SQL Formatter](https://github.com/sql-formatter-org/sql-formatter) -> `npx sql-formatter db/setup/schema.sql`

The only change is the addition of this field.

**Comment by Damrod** [comment]
Also please, document how to regenerate this transcendence-app/swagger-spec.json file

**Comment by gbudau** [comment]
Good point :+1:
This file is generated automatically by the transcendence-app at startup whenever the OpenAPI spec changes.

File: transcendence-app/src/user/dto/user.dto.ts [hunk 2]
```diff
@@ -9,11 +10,17 @@ import {
 export class UserDto {
   @IsString()
   @IsNotEmpty()
+  @Transform(({ value }: TransformFnParams) => value?.trim())
```
Comment (gbudau) [inline transcendence-app/src/user/dto/user.dto.ts:13]: This decorator disallows assigning strings containing only whitespace.

**Review by Damrod — APPROVED** [review]
I thought I approved this, but apparently I didn't. Now

**Comment by gbudau** [comment]
Thank you!

**Pull request merged**